lib/Data/Database.php

```diff
@@ -233,37 +252,41 @@ public function read($pasteid)
         }
 
         self::$_cache[$pasteid] = false;
+        $rawData                = '';
         try {
             $paste = self::_select(
                 'SELECT * FROM ' . self::_sanitizeIdentifier('paste') .
                 ' WHERE dataid = ?', array($pasteid), true
             );
+            if ($paste !== false) {
+                $rawData = self::$_type === 'oci' ? self::_clob($paste['DATA']) : $paste['data'];
+            }
         } catch (Exception $e) {
             $paste = false;
         }
         if ($paste === false) {
             return false;
         }
         // create array
-        $data       = Json::decode($paste['data']);
+        $data       = Json::decode($rawData);
         $isVersion2 = array_key_exists('v', $data) && $data['v'] >= 2;
         if ($isVersion2) {
             self::$_cache[$pasteid] = $data;
             list($createdKey)       = self::_getVersionedKeys(2);
         } else {
-            self::$_cache[$pasteid] = array('data' => $paste['data']);
+            self::$_cache[$pasteid] = array('data' => $paste[self::_sanitizeColumn('data')]);
             list($createdKey)       = self::_getVersionedKeys(1);
         }
 
         try {
-            $paste['meta'] = Json::decode($paste['meta']);
+            $paste['meta'] = Json::decode($paste[self::_sanitizeColumn('meta')]);
         } catch (Exception $e) {
             $paste['meta'] = array();
         }
         $paste                                       = self::upgradePreV1Format($paste);
         self::$_cache[$pasteid]['meta']              = $paste['meta'];
-        self::$_cache[$pasteid]['meta'][$createdKey] = (int) $paste['postdate'];
-        $expire_date                                 = (int) $paste['expiredate'];
+        self::$_cache[$pasteid]['meta'][$createdKey] = (int) $paste[self::_sanitizeColumn('postdate')];
```

Comment:
Why do you need to sanitize all that here (and below) now and did not need to do that before?

After all, it is just about reading/saving the data from the SQL DB into the cache here, no SQL injection problem/sanitation required, is it?

Reply:
Ahhhh looking at the `_sanitizeColumn` function now I see why. However, do we really need to do that in cache too and complicate the code here?

Reply:
I tried to invert the logic and generalize it directly into the `_select()`, so that the rest of our code can remain the same and the returned keys are all lowercase as we expect:
https://github.com/PrivateBin/PrivateBin/commit/2182cdd44f092277ce829cb9effd861ceb415c06#diff-c4661c92ca75850dcd481b990c4a8c40b5203ad07a25f128f709101aefec6216R589